src/profile-logic/process-profile.js

```diff
@@ -560,13 +604,30 @@ function _processSamples(geckoSamples: GeckoSampleStruct): SamplesTable {
   };
 }
 
+// Converts a schema array-based extension list to a list of Extension
+// objects with the same information.
+function _processExtensions(extensions: ExtensionMeta): Extension[] {
+  const result: Extension[] = [];
+
+  const { schema, data } = extensions;
+  for (const ext of data) {
+    result.push({
+      id: ext[schema.id],
+      name: ext[schema.name],
+      baseURL: ext[schema.baseURL],
+    });
+  }
+  return result;
+}
+
 /**
  * Convert the given thread into processed form. See docs/gecko-profile-format for more
  * information.
  */
 function _processThread(
```

Comment:
Since we're changing our assumptions about the Gecko profile format, a few things should also change:

* [Our docs on the Gecko format should be updated](https://github.com/devtools-html/perf.html/blob/master/docs/gecko-profile-format.md#source-data-format)
* We should have a test for the upgrade process. Something like this:

```diff
diff --git a/src/test/unit/profile-data.test.js b/src/test/unit/profile-data.test.js
index 6fba21e2..5670f0a2 100644
--- a/src/test/unit/profile-data.test.js
+++ b/src/test/unit/profile-data.test.js
@@ -293,6 +293,16 @@ describe('process-profile', function() {
       expect(profile.threads.length).toEqual(3);
     });
   });
+  describe('extensions metadata', function() {
+    const geckoProfile = getGeckoProfile();
+    // Modify profile to have extension data.
+    const profile = processProfile(geckoProfile);
+    expect(profile.meta.extensions).toEqual(
+      {
+        // Fill out here ...
+      }
+    );
+  });
 });
 
 describe('profile-data', function() {
````